packages/pwa-kit-extension-sdk/src/types.ts

```diff
@@ -46,17 +46,27 @@ export interface GetRoutesParams {
     locals: Record<string, any>
 }
 
-/*
- * Same as react-router's RouteProps, but now supports `componentName` if the route is meant to be serializable
- */
-export type RouteProps = _RouteProps & {
-    componentName?: string
-}
-
 /**
  * Parameters for the `beforeRouteMatch` method
  */
 export interface BeforeRouteMatchParams {
     allRoutes: RouteProps[]
     locals: Record<string, any>
 }
+
+/**
+ * This type is used in getComponentMap() to map the component name to the component itself
+ * when deserializing an extension.
+ */
+export type ComponentMap = {
+    [key: string]: React.ComponentType<any>
```

Comment:
Is you want to get super specific about this type you might want to do something like this:

```
type ComponentMap = {
  [key in ReturnType<ApplicationExtension['getName']>]: React.ComponentType<any>
};
```